packages/plugins/graphql/server/src/services/builders/resolvers/association.ts

```diff
@@ -57,16 +57,21 @@ export default ({ strapi }: Context) => {
             auth,
           }
         );
-
         const transformedQuery = strapi.get('query-params').transform(targetUID, sanitizedQuery);
 
         const defaultFilters = {
           where: {
-            // Return the same draft and publish version as the parent
-            publishedAt: { $notNull: 'publishedAt' in parent ? parent.publishedAt !== null : true },
+            publishedAt: {
+              $notNull: context.rootQueryArgs?.status
+                ? // Filter by the same status as the root query if the argument is present
+                  context.rootQueryArgs?.status === 'published'
+                : // Otherwise fallback to the published version
+                  true,
```

Comment:
I can see some users wanting to only select published items at the top level, but include the draft relations in children.

So I think we should use:

local arg if present -> rootArg if present -> published by default